### DISCUSSION TITLE
Envkey With Parcel 2

### DISCUSSION THREAD

**Discussion by imakeinterfaces** [envkey/envkey#14] [Q&A] [answered]
Has anyone found a proper way of using envkey with parcel 2? I have yet to find a good way to do this.

**Comment by danenania** [comment, marked as answer]
I'm not too familiar with parcel, but looking at [this issue](https://github.com/parcel-bundler/parcel/issues/3965), it seems that parcel allows you to pass through any environment variables you set when running the build by referencing them as `process.env.VARIABLE`. So from the example given in that issue:

```
<!-- index.html -->
<script src="index.js"></script>
```
```
// index.js
window.FOO = process.env.FOO;
```
```
FOO=abc parcel build index.html
```

Given this, I think it should be trivial to use with [envkey-source](https://docs-v2.envkey.com/docs/envkey-source). You could just do the following:

```
envkey-source -- parcel build index.html
```

And then reference whatever vars you need on `process.env` in your app. Just make sure you don't reference anything sensitive... I'd even suggest separating your non-sensitive frontend vars into a different app so there's no danger of this happening (and using [blocks](https://docs-v2.envkey.com/docs/blocks) if needed to avoid duplication).

Does that help?

**Reply by imakeinterfaces** [reply]
The easiest way to do this is with the package envkey-shell https://www.npmjs.com/package/envkey-shell
This way there isn't a need to side-load a package that's not in npm. 

I don't know if there is a security drawback in using this method. Are the values going to be obfuscated on the front end using this method?

**Reply by imakeinterfaces** [reply]
It seems that nextjs has a similar issue while there is no nodejs package (outside of the cli wrapper)
